clients/filesystem-fuse/src/filesystem.rs

```diff
@@ -239,3 +352,406 @@ pub trait FileWriter: Sync + Send {
+                file_stat.set_file_id(parent_file_id, self.next_file_id());
+                file_manager.insert(file_stat.parent_file_id, file_stat.file_id, &file_stat.path);
+            }
+            Some(file) => {
+                // use the exist file id
+                file_stat.set_file_id(file.parent_file_id, file.file_id);
+            }
+        }
+    }
+
+    async fn open_file_internal(
+        &self,
+        file_id: u64,
+        flags: u32,
+        kind: FileType,
+    ) -> Result<FileHandle> {
+        let file_entry = self.get_file_entry(file_id).await?;
+
+        let mut file = {
+            match kind {
+                FileType::Directory => {
+                    self.fs
+                        .open_dir(&file_entry.file_name, OpenFileFlags(flags))
+                        .await?
+                }
+                FileType::RegularFile => {
+                    self.fs
+                        .open_file(&file_entry.file_name, OpenFileFlags(flags))
+                        .await?
+                }
+                _ => return Err(Errno::from(libc::EINVAL)),
+            }
+        };
+        // set the exists file id
+        file.set_file_id(file_entry.parent_file_id, file_id);
+        let file = self.opened_file_manager.put_file(file);
+        let file = file.lock().await;
+        Ok(file.file_handle())
+    }
+}
+
+#[async_trait]
+impl<T: PathFileSystem> RawFileSystem for SimpleFileSystem<T> {
+    async fn init(&self) -> Result<()> {
+        // init root directory
+        self.file_entry_manager.write().await.insert(
+            Self::ROOT_DIR_PARENT_FILE_ID,
+            Self::ROOT_DIR_FILE_ID,
+            Self::ROOT_DIR_NAME,
+        );
+        self.fs.init().await
+    }
+
+    async fn get_file_path(&self, file_id: u64) -> String {
+        let file = self.get_file_entry(file_id).await;
+        file.map(|x| x.file_name).unwrap_or_else(|_| "".to_string())
+    }
+
+    async fn valid_file_handle_id(&self, file_id: u64, fh: u64) -> Result<()> {
+        let fh_file_id = self
+            .opened_file_manager
+            .get_file(fh)
+            .ok_or(Errno::from(libc::EBADF))?
+            .lock()
+            .await
+            .file_stat
+            .file_id;
+
+        (file_id == fh_file_id)
+            .then_some(())
+            .ok_or(Errno::from(libc::EBADF))
+    }
+
+    async fn stat(&self, file_id: u64) -> Result<FileStat> {
+        let file = self.get_file_entry(file_id).await?;
+        let mut stat = self.fs.stat(&file.file_name).await?;
+        stat.set_file_id(file.parent_file_id, file.file_id);
+        Ok(stat)
+    }
+
+    async fn lookup(&self, parent_file_id: u64, name: &str) -> Result<FileStat> {
+        let parent_file = self.get_file_entry(parent_file_id).await?;
+        let mut stat = self.fs.lookup(&parent_file.file_name, name).await?;
+        // fill the file id to file stat
+        self.fill_file_id(&mut stat, parent_file_id).await;
```

Comment:
`fill_file_id` is confusing, you changed too many things in one function, suggest to rewrite  something like
```rust
    let id = self.try_update_file_mapping(stat.file_name, parent_file_id).await
    stat.file_id = id
```

Reply:
Changed the function name

Reply:
`resolve_file_id_to_filestat` means fill id to the stat, but the content is  far beyond  the name

Reply:
I think this name is already clear. If there's a file_id, use it; if not, request new one.